firefox-ios/Client/Frontend/Home/Homepage Rebuild/HomepageDiffableDataSource.swift

```diff
@@ -38,21 +49,20 @@ final class HomepageDiffableDataSource:
         }
     }
 
-    func applyInitialSnapshot(state: HomepageState) {
+    func updateSnapshot(state: HomepageState) {
```

Comment:
@dataports this is my implementation of header reloading without explicitly calling `reloadSections` or using the generated UUID. This allows the Pocket section's header to be appropriately updated because it is diffing on the `sectionHeaderState`. When the font color changes we need to reload the header but not every time `updateSnapshot` is called.

@FilippoZazzeroni I profiled this with this implementation and did not see any hangs. For completeness, I ran instruments with the UUID implementation similar to the `TabDisplayDiffableDataSource`. I was able to reproduce the hangs with that implementation. This indicates to me that the hangs you identified in `TabDisplayDiffableDataSource` are because we are forcing a reload every time we call `updateSnapshot` with a new UUID, not necessarily because `updateSnapshot` is being called multiple times. 

Let me know your thoughts!

Reply:
Ok thanks so much for the insight @Cramsden then i'll dig into why we are passing a new UUID all the times.

Reply:
@FilippoZazzeroni My understanding was that we needed the inactive tabs header to reload. It is a bit weird to force that with diffable data sources and calling `relaodSections` on an empty section crashes on iOS 15. [Here](https://github.com/mozilla-mobile/firefox-ios/pull/22961) is the previous PR to fix the crash. Possibly we can just listen to the data we care about changing to reload the inactive tabs instead of doing it always with the forced UUID. I bet @dataports has more insight on when we need to reload the inactive tabs header.

Reply:
Yes i agree @Cramsden that would be better to just listen to the correct data, instead of forcing reload. I'll have a look if i can find some solutions to that, also right now with the state fix in the hangs PR it apply snapshot should be called once, but still forcing it.

Reply:
Ahhh this implementation makes sense to me. Basically if we want state updates in "header" cells we could just make them `cells` in DiffableDatasource instead of using the supplementary views because the supplementary views aren't really intended to be changed that way. We could also probably use this approach with the `SectionHeader` view on the Tab Tray right now and it would be a bit cleaner I think. Curious what @cyndichin thinks when she gets back too.

 @FilippoZazzeroni  if you have time, feel free to play around with implementing this or something else on that tab tray code that's using UUID currently, I know it's not ideal right now. 😭

Reply:
Definitely agree this is the route we want to go instead of the UUID. In this case, we are reloading specifically when there is a change in a specific font color. The UUID was a temporary fix, but seems like that we encountered issues. 

@dataports In terms of the tab tray work, we can follow similar by finding a property that is unique to the section and when it changes it causes the section to reload (similar to how we have the text color here). 

Thinking off the top of my head, I believe the issue was with the animation not reloading when showing / hiding the inactive tabs section, so maybe we can have a state that has a bool value `showInactiveTabs`, which would reload section. 

On a separate note, related to this PR, @Cramsden will there be a case where text color is nil? Or should we add the default case and make the associated value a non nil type?

Reply:
thanks all for chiming in on this!